### PR TITLE
feat(shields): Add splitkb.com Aurora Lily58.

### PR DESCRIPTION
**Pull request by petejohanson**
## Board/Shield Check-list
 - [x] This board/shield is tested working on real hardware
 - [x] Definitions follow the general style of other shields/boards upstream ([Reference](https://zmk.dev/docs/development/new-shield))
 - [x] `.zmk.yml` metadata file added
 - [x] Proper Copyright + License headers added to applicable files (Generally, we stick to "The ZMK Contributors" for copyrights to help avoid churn when files get edited)
 - [x] General consistent formatting of DeviceTree files
 - [x] Keymaps do not use deprecated key defines (Check using the [upgrader tool](https://zmk.dev/docs/codes/keymap-upgrader))
 - [x] `&pro_micro` used in favor of `&pro_micro_d/a` if applicable
 - [x] If split, no name added for the right/peripheral half
 - [x] Kconfig.defconfig file correctly wraps *all* configuration in conditional on the shield symbol
 - [x] `.conf` file has optional extra features commented out
 - [x] Keyboard/PCB is part of a shipped group buy or is generally available in stock to purchase (OSH/personal projects without general availability should create a zmk-config repo instead)

### CODE REVIEW

**Comment by laureyn** [comment]
This works on my keyboard I just built using the lily58 kit from splitkb using a nice!nano v2.
Underglow is also working out of the box.

File: app/boards/shields/splitkb_aurora_lily58/splitkb_aurora_lily58.dtsi [hunk 1]
```diff
@@ -0,0 +1,78 @@
...
+/ {
+
+	chosen {
+		zephyr,display = &oled;
+		zmk,matrix_transform = &default_transform;
+	};
+
+	default_transform: keymap_transform_0 {
+		compatible = "zmk,matrix-transform";
+		columns = <16>;
```
Comment (caksoylar) [inline app/boards/shields/splitkb_aurora_lily58/splitkb_aurora_lily58.dtsi:18]:
```suggestion
		columns = <14>;
```
I think this doesn't do anything at the moment but should this be 14?

File: app/boards/shields/splitkb_aurora_lily58/splitkb_aurora_lily58.keymap [hunk 2]
```diff
@@ -0,0 +1,70 @@
...
+		};
+
+		lower_layer {
+// ------------------------------------------------------------------------------------------------------------
+// | BTCLR | BT1 | BT2 |  BT3 |  BT4 |  BT5 |                   |      |       |       |      |       |       |
+// |  F1   |  F2 |  F3 |  F4  |  F5  |  F6  |                   |  F7  |  F8   |  F9   |  F10 |  F11  |  F12  |
+// |   `   |  !  |  @  |  #   |  $   |  %   |                   |  ^   |  &    |  *    |  (   |   )   |   ~   |
+// |       |     |     |      |      |      |        |  |       |      |  _    |  +    |  {   |   }   |  "|"  |
+//                     |      |      |      |        |  |       |      |       |       |
+			bindings = <
```
Comment (caksoylar) [inline app/boards/shields/splitkb_aurora_lily58/splitkb_aurora_lily58.keymap:41]: Nit: might want to prefer non `KP_` versions for `KP_ASTERISK` (`STAR`) and `KP_PLUS` (`PLUS`, also in raise layer). The alignment would need adjustment as well after these replacements.

File: app/boards/shields/splitkb_aurora_lily58/splitkb_aurora_lily58.conf [hunk 3]
```diff
@@ -0,0 +1,9 @@
+# Uncomment these two line to add support for encoders to your firmware
+# CONFIG_EC11=y
+# CONFIG_EC11_TRIGGER_GLOBAL_THREAD=y
+
+# Uncomment the following line to enable the Kyria OLED Display
```
Comment (caksoylar) [inline app/boards/shields/splitkb_aurora_lily58/splitkb_aurora_lily58.conf:5]:
```suggestion
# Uncomment the following line to enable the OLED Display
```

Comment (caksoylar) [reply app/boards/shields/splitkb_aurora_lily58/splitkb_aurora_lily58.conf:5]: Can replace with `Lily58`, also just noticed the same with the Aurora Corne shield.

File: app/boards/shields/splitkb_aurora_lily58/splitkb_aurora_lily58.dtsi (hunk 1 above)
Comment (petejohanson) [reply app/boards/shields/splitkb_aurora_lily58/splitkb_aurora_lily58.dtsi]: Indeed. Fixed.

File: app/boards/shields/splitkb_aurora_lily58/splitkb_aurora_lily58.keymap (hunk 2 above)
Comment (petejohanson) [reply app/boards/shields/splitkb_aurora_lily58/splitkb_aurora_lily58.keymap:41]: Agreed. Replaced.

File: app/boards/shields/splitkb_aurora_lily58/splitkb_aurora_lily58.conf (hunk 3 above)
Comment (petejohanson) [reply app/boards/shields/splitkb_aurora_lily58/splitkb_aurora_lily58.conf]: Good catch. Fixed.

**Pull request merged**